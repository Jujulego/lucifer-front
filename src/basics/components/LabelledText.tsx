import React, { ReactNode } from 'react';
import clsx from 'clsx';

import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import { StyledProps } from 'utils/style';

// Styles
const useStyles = makeStyles({
  root: {
    position: 'relative',
    width: '100%',

    display: 'inline-flex',
    alignItems: 'center'
  },
  label: {
    position: 'absolute',
    top: -6,
    left: 14,
  },
  content: {
    width: '100%',
    padding: '18.5px 14px',

    fontSize: '1rem',
    lineHeight: '1.1876em',
  },
  endAdorned: {
    paddingRight: 0
  }
});

// Types
export type LabelledTextClassKey = 'root' | 'label' | 'content' | 'endAdornment';
export type LabelledTextProps = StyledProps<LabelledTextClassKey> & {
  label: string;
  endAdornment?: ReactNode;
  children: ReactNode;
}

// Component
const LabelledText = React.memo((props: LabelledTextProps) => {
  const {
    label, endAdornment,
    children
  } = props;

  // Render
  const styles = useStyles(props);

  return (
    <div className={styles.root}>
      <Typography className={styles.label} variant='caption'>{ label }</Typography>
      <div className={clsx(styles.content, { [styles.endAdorned]: !!endAdornment })}>
        { children }
      </div>
      { endAdornment }
    </div>
  );
});

export default LabelledText;
