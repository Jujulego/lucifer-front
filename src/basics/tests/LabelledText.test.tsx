import React from 'react';

import { createShallow } from '@material-ui/core/test-utils';
import { Check as CheckIcon } from '@material-ui/icons';

import LabelledText from '../components/LabelledText';

// Setup
let shallow: ReturnType<typeof createShallow>;

beforeAll(() => {
  shallow = createShallow();
});

// Tests
it('should render correctly', () => {
  // Render
  const wrapper = shallow(
    <LabelledText label="label">Content</LabelledText>
  );

  // Checks
  expect(wrapper).toMatchSnapshot();
});

it('should render with adornment', () => {
  // Render
  const wrapper = shallow(
    <LabelledText
      label="label"
      endAdornment={<CheckIcon />}
    >
      Content
    </LabelledText>
  );

  // Checks
  expect(wrapper).toMatchSnapshot();
});
