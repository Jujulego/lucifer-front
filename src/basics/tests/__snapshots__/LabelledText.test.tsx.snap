// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-2"
    variant="caption"
  >
    label
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-content-3"
  >
    Content
  </div>
</div>
`;

exports[`should render with adornment 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-2"
    variant="caption"
  >
    label
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-content-3 makeStyles-endAdorned-4"
  >
    Content
  </div>
  <CheckIcon />
</div>
`;
